packages/api/src/base/Init.ts

```diff
@@ -94,33 +98,59 @@ export default abstract class Init<ApiType> extends Decorate<ApiType> {
       this._rpcCore.chain.getRuntimeVersion().toPromise()
     ]);
 
-    // based on the node, inject specific types
-    this.registerTypes(
-      SPEC_TYPES[this._runtimeVersion.specName.toString()]
-    );
-
     const metadataKey = `${this._genesisHash}-${(this._runtimeVersion as RuntimeVersion).specVersion}`;
     const metadata = metadataKey in optMetadata
       ? new Metadata(optMetadata[metadataKey])
       : await this._rpcCore.state.getMetadata().toPromise();
 
+    // based on the node, inject specific types - this is very specific to known chains
+    this.registerTypes(SPEC_TYPES[this._runtimeVersion.specName.toString()]);
+
+    // HACK-ish Old EventRecord format for e.g. Alex, based on metadata versions
+    if (metadata.version <= 3) {
+      this.registerTypes(TYPES_SUBSTRATE_1);
+    }
+
     // get unique types & validate
     metadata.getUniqTypes(false);
 
+    // // subscribe to the runtime version, updating metadata as changes are made
+    this._versionSub = this.runtimeVersionObservable().subscribe();
+
     return metadata;
   }
 
-  private async initFromMeta (metadata: Metadata): Promise<boolean> {
-    // HACK-ish Old EventRecord format for e.g. Alex, based on \metadata format
-    if (metadata.version <= 3) {
-      this.registerTypes(TYPES_SUBSTRATE_1);
-    }
+  private runtimeVersionObservable (): Observable<Promise<boolean>> {
```

Comment:
`Observable<Promise<boolean>>` is kind of weird, why not just Observable?

Reply:
Yes, exceptionally weird. First order of business is actually getting it to work since it is the last thing remaining for 0.90 - actually haven’t figured out how this _should_ look atm, more or less going through steps to see what works and what doesn’t. 

So atm -

- have subscriptions fixed
- actually get events on UI and API on upgrades (not a pleasant testing process at all)
- have some really weird horrors in block decoding (after update)

Hopefully can get some sort of semblance of working tomorrow and start looking how to best structure after that.

Reply:
EDIT: to clarify, depending on the testing, this may not make 0.90, but needs to be in before a first runtime upgrade in Kusama (latest 2 weeks delay after 0.9). 

So trying, but as mentioned this one is a real logistics nightmare (bumping, re-compilation, disabling modules, and the verifying against actual use, etc), so doing what can be done